pkg/controllers/resources/pods/syncer.go

```diff
@@ -545,3 +596,102 @@ func (s *podSyncer) getNodeIP(ctx *synccontext.SyncContext, name string) (string
 
 	return nodeService.Spec.ClusterIP, nil
 }
+
+func (s *podSyncer) ExcludeVirtual(obj client.Object) bool {
+	pod, ok := obj.(*corev1.Pod)
+	if !ok {
+		return false
+	}
+
+	return s.excludeVirtualPodOnRuntime(pod) || s.excludeVirtualPodOnPriority(pod)
+}
+
+func (s *podSyncer) excludeVirtualPodOnRuntime(pod *corev1.Pod) bool {
+	if pod.Spec.RuntimeClassName == nil || *pod.Spec.RuntimeClassName == "" && selector.IsLabelSelectorEmpty(s.runtimeClassLabelSelector) {
+		return false
+	}
+
+	runtimeClass, err := s.physicalClusterClient.NodeV1().RuntimeClasses().
+		Get(context.Background(), *pod.Spec.RuntimeClassName, metav1.GetOptions{})
+	if err != nil {
+		klog.FromContext(context.Background()).Info(
+			fmt.Sprintf("Warning: Pod %q will not be synced to host cluster, because RuntimeClass %q couldn't be found: %v", pod.Name, *pod.Spec.RuntimeClassName, err))
+		return true
+	}
+
+	exclude := !selector.StandardLabelSelectorMatches(runtimeClass, s.runtimeClassLabelSelector)
+	if exclude {
+		klog.FromContext(context.Background()).Info(
+			fmt.Sprintf("Warning: Pod %q will not be synced to host cluster, because RuntimeClass %q does NOT match the label selector in the 'sync.fromHost.runtimeClasses' configuration", pod.Name, *pod.Spec.RuntimeClassName))
+	}
+	return exclude
+}
+
+func (s *podSyncer) excludeVirtualPodOnPriority(pod *corev1.Pod) bool {
+	if pod.Spec.PriorityClassName == "" && selector.IsLabelSelectorEmpty(s.priorityClassLabelSelector) {
+		return false
+	}
+
+	priorityClass, err := s.physicalClusterClient.SchedulingV1().PriorityClasses().
+		Get(context.Background(), pod.Spec.PriorityClassName, metav1.GetOptions{})
+	if err != nil {
+		klog.FromContext(context.Background()).Info(
+			fmt.Sprintf("Warning: Pod %q will not be synced to host cluster, because PriorityClass %q couldn't be found: %v", pod.Name, pod.Spec.PriorityClassName, err))
+		return true
+	}
+
+	exclude := !selector.StandardLabelSelectorMatches(priorityClass, s.priorityClassLabelSelector)
+	if exclude {
+		klog.FromContext(context.Background()).Info(
+			fmt.Sprintf("Warning: Pod %q will not be synced to host cluster, because PriorityClass %q does NOT match the label selector in the 'sync.fromHost.priorityClasses' configuration", pod.Name, pod.Spec.PriorityClassName))
+	}
+	return exclude
+}
+
+func (s *podSyncer) ExcludePhysical(_ client.Object) bool {
+	return false
```

Comment:
This implicitly not checks this anymore https://github.com/loft-sh/vcluster/blob/main/pkg/syncer/syncer.go